platform/features/layout/src/LayoutController.js

```diff
@@ -127,6 +127,25 @@ define(
                 e.preventDefault();
             }
 
+            function getComposition(domainObject){
+                return domainObject.useCapability('composition');
+            }
+
+            function composeView (composition){
+                $scope.composition = composition;
+                return composition.map(function (object) {
+                        return object.getId();
+                    }) || [];
+            }
+
+            //Will fetch fully contextualized composed objects, and populate
+            // scope with them.
+            function refreshComposition() {
+                return getComposition($scope.domainObject)
+                    .then(composeView)
+                    .then(self.layoutPanels);
+            }
```

Comment:
`refreshComposition` could have `ids` available to it as an argument from the watch - could simplify code a little bit and remove some of the above functions if we used these to `layoutPanels` instead of mapping them back from domain objects in `composeView`.

There's also a race condition introduced here, since `composition` promises will not necessarily resolve in the same order they are requested. For instance, consider the following sequence:

* A layout is loaded containing composition A, B, C. A request is issued but it doesn't resolve right away (for whatever reason)
* From the tree, the user creates a Link, adding D to the composition of the layout.
* A new `refreshComposition` call loads A, B, C, D. This happens to resolve quickly; A, B, C, D are placed in scope.
* The old composition promise finally resolves; A, B, C is put into scope.
* Digest gets triggered, user sees A, B, C, thinks "where is that D that I just added?"

There's [some logic in `mct-representation` that prevents this](https://github.com/nasa/openmctweb/blob/560454e7c253b96c0bf461d0c2c7e336be17f560/platform/representation/src/MCTRepresentation.js#L127-L132) for capabilities loaded via `uses`, so this would be a regression.